src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -0,0 +1,214 @@
+                
+                // Return 200 OK for approved review and 201 for review in pending status
+                if (firstReleaseStatusOnly != true && latestAutomaticApiRevisions.IsApproved)
+                {
+                    return Ok();
+                }
+                else 
+                {
+                    if (review.IsApproved)
+                    {
+                        return StatusCode(statusCode: StatusCodes.Status201Created);
+                    }
+                    // Return 202 to indicate package name is not approved
+                    return StatusCode(statusCode: StatusCodes.Status202Accepted);
+                }
+            }
+            throw new Exception("Review is not found for package " + packageName);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> CreateApiReview(
+            string buildId,
+            string artifactName,
+            string originalFilePath,
+            string reviewFilePath,
+            string label,
+            string repoName,
+            string packageName,
+            bool compareAllRevisions,
+            string project
+            )
+        {
+            using var memoryStream = new MemoryStream();
+            var codeFile = await _codeFileManager.GetCodeFileAsync(repoName: repoName, buildId: buildId, artifactName: artifactName,
+                packageName: packageName, originalFileName: originalFilePath, codeFileName: reviewFilePath, originalFileStream: memoryStream,
+                project: project);
+
+            var apiRevision = await CreateAutomaticRevisionAsync(codeFile: codeFile, label: label, originalName: originalFilePath, memoryStream: memoryStream, compareAllRevisions);
+            if (apiRevision != null)
+            {
+                var reviewUrl = $"{this.Request.Scheme}://{this.Request.Host}/Assemblies/Review/{apiRevision.ReviewId}?revisionId={apiRevision.Id}";
+                return apiRevision.IsApproved ? Ok(reviewUrl) : StatusCode(statusCode: StatusCodes.Status201Created, reviewUrl);
+            }
+            // Return internal server error for any unknown error
+            return StatusCode(statusCode: StatusCodes.Status500InternalServerError);
+        }
+
+        private async Task<APIRevisionListItemModel> CreateAutomaticRevisionAsync(CodeFile codeFile, string label, string originalName, MemoryStream memoryStream, bool compareAllRevisions = false)
+        {
+            var createNewRevision = true;
+            var review = await _reviewManager.GetReviewAsync(packageName: codeFile.PackageName, language: codeFile.Language, isClosed: null);
+            var apiRevision = default(APIRevisionListItemModel);
+            var renderedCodeFile = new RenderedCodeFile(codeFile);
+            var apiRevisions = (await _apiRevisionsManager.GetAPIRevisionsAsync(review.Id)).OrderByDescending(r => r.CreatedOn).ToList();
```

Comment:
`review` will be null when automatic API review request comes for a brand-new package and this line will fail due to null.
`review.Id`

Reply:
I have moved this into the `apiRevision .Any()` check